DFJR-719 - Remove prototype code

richaagarwal opened the pull request:
"Inside our atomic elements templates (/_includes/ folder) and page templates in the we want to remove all references to prototype stock content." 

## Review

- @kave 
- @rosskarchner 
- @kurtw 
- @jimmynotjim 
- @sebworks 
- @KimberlyMunoz 
- @anselmbradford 

## Notes

- I can't run `gulp test:acceptance` yet :( but did do a visual test on localhost

kave commented:
looks good, since we are removing these prototypes. There are prototype urls in the `urls.py` pertaining to each page template that we can now also remove in this PR

richaagarwal commented:
@kave done!

anselmbradford commented on cfgov/jinja2/v1/browse-basic/index.html:
```diff
@@ -51,49 +29,29 @@
+            {% if 'related_posts' in block.block_type %}
+                {{ related_posts.render(block) }}
+            {% else %}
+                {% if 'heading' in block.block_type %}
```
Doesn't need to hold this up, but `if` statement can be combined with `else` statement above (turned into an `elif`).

anselmbradford commented on cfgov/jinja2/v1/landing-page/index.html:
```diff
@@ -90,32 +42,27 @@
+            {% if 'related_posts' in block.block_type %}
+                {{ related_posts.render(block) }}
+            {% else %}
+                {% if 'heading' in block.block_type %}
```
`if` can be combined with above `else` here too.

kurtw replied:
bump

richaagarwal replied:
@anselmbradford @kurtw Sorry, forgot to comment on this earlier.  It seemed a little out of scope and I wanted to keep this PR as tight as possible since it touches a lot of code. What do you think?

anselmbradford commented on cfgov/jinja2/v1/learn-page/index.html:
```diff
@@ -224,49 +40,27 @@
+        {% if 'related_posts' in block.block_type %}
+            {{ related_posts.render(block) }}
+        {% else %}
+            {% if 'heading' in block.block_type %}
```
`if` can be combined with above `else`

anselmbradford commented on cfgov/jinja2/v1/sublanding-page/index.html:
```diff
@@ -129,51 +46,28 @@
+        {% elif 'email_signup' in block.block_type %}
+            {{ render_stream_child(block) }}
+        {% else %}
+            {% if 'heading' in block.block_type %}
```
`if` can be combined with `else`

kurtw commented on cfgov/jinja2/v1/_includes/molecules/text-introduction.html:
```diff
@@ -18,26 +18,6 @@
-        })
-    %}
-{% endif %}
 {% set value = value or global_dict.text_introduction %}
```
This should also be removed.

kurtw commented on cfgov/jinja2/v1/_includes/organisms/full-width-text.html:
```diff
@@ -11,69 +11,6 @@
-        ]
-    }) %}
-{% endif %}
 {% set value = value or global_dict.full_width_text %}
```
This should be removed.

kurtw commented on cfgov/jinja2/v1/_includes/organisms/image-text-25-75-group.html:
```diff
@@ -1,19 +1,3 @@
-        })
-    %}
-{% endif %}
 {% set value = value or global_dict.image_text_25_75 %}
```
This should be removed.

richaagarwal replied:
@kurtw I'm not sure I follow - the section you've commented on is removed in this PR. Is there something else that should be removed?

kave commented:
@richaagarwal are you able to run `gulp test:acceptance` yet?

richaagarwal commented:
@kave no, I still get the same error I was getting before `Error: Cannot find module 'lodash/object/assign'`

sebworks commented:
@richaagarwal, please pull down the latest.

richaagarwal commented:
@sebworks I have, unfortunately :\

richaagarwal commented:
I get the same error on `gulp build` as well.

kurtw commented on cfgov/jinja2/v1/_includes/organisms/well.html:
```diff
@@ -14,10 +14,6 @@
-    {% set lipsumText = lipsum(1, false, 10, 25) | safe %}
-    {% do global_dict.update({'well': { 'content' :  lipsumText  }}) %}
-{% endif %}
 {% set value = value or global_dict.well %}
```
This should be removed too. Any instance of this should be removed in the codebase for the time being.

richaagarwal replied:
Got it. Okay there's a bunch more prototype code I'm seeing by following that. Stay tuned!

richaagarwal commented:
This is a WIP for now. I'm getting some errors now on value not being defined, so I'm not sure if we're incorrectly relying on them from the prototype code or if I got rid of things I shouldn't have.

richaagarwal commented on cfgov/jinja2/v1/document-detail/index.html:
```diff
@@ -91,7 +91,6 @@
 
                 })
             %}
-            {% set value = global_dict.expandable %}
```
@kave @kurtw Should I be getting rid of instances like this?

richaagarwal commented:
@KimberlyMunoz @anselmbradford @jimmynotjim @sebworks When running acceptance tests, I get an error on a couple pages saying 'value' is undefined. I've gotten rid of all instances of setting value based on the global dictionary, so I'm not sure if this is expected or something else needs to be resolved. 

I see this on `/the-bureau/about-deputy-director/` and `/budget`. See attached screenshot. 
<img width="1180" alt="screen shot 2016-02-04 at 11 45 37 am" src="https://cloud.githubusercontent.com/assets/354591/12827596/559ef876-cb35-11e5-919a-0b0df76e9d33.png">

jimmynotjim commented:
I'm seeing 47 failures on the `flapjack` branch, so I'd say lets merge this as is, and we can push the acceptance test issue in Jira up a level. Once the home page is merged I'll have time if no one else gets to it first.

kurtw commented:
Thanks for that @jimmynotjim

jimmynotjim commented:
Now checking your branch, and I'm seeing value is undefined errors all over, including the home page, so lets work through those (unrelated to acceptance test issues)

richaagarwal commented:
I think is ready for a final review!

kurtw commented:
Other than Ans' comments about combining the nested conditionals, are those jinja2 files that still have global_dict defined in them confirmed as necessary?

jimmynotjim commented:
`/the-bureau/` and `/careers/application-process/` still need fixing. Looks like this time it's the hero, which is due to the fact that these are old pages from outside the CMS that have to have a value passed in. I think if you re-instate the values [here](https://github.com/cfpb/cfgov-refresh/pull/1393/files#diff-58b44630c1dfe58a4b807321002b0cb0L53) and [here](https://github.com/cfpb/cfgov-refresh/pull/1393/files#diff-954d26b6ea36230d174d1989f59afd9bL48) this can be merged. Maybe just add `{# TODO: #}` notes so we don't lose track of them.

richaagarwal commented:
@jimmynotjim I fixed `/the-bureau/` and `/careers/application-process/` by passing hero in as a macro. The only issue is I'm not sure how to pass the image URL into the photo filter (`{% set photo=image(value.image.upload, 'original') %}` in `molecules/hero.html`), so I've added that as a TODO.  It wants an AbstractImage but I don't know where that magic should happen. Any ideas?

richaagarwal commented:
@kurtw I'm going through the global_dict instances now. But for something like this:
```            <div class="m-full-width-text
                        {{ 'm-full-width-text__serif' if
                        global_dict.full_width_serif else '' }}">
```
it seems like we want to replace that logic some other way? What are your thoughts there? This is in `organisms/full-width-text.html`.

richaagarwal commented:
@jimmynotjim after my latest clean-up, I'm getting errors (on the homepage again) because `page` is undefined. It seems like we're still relying on it in one of the molecules I updated here: https://github.com/cfpb/cfgov-refresh/commit/16847b3cb0c9b19c45b62b43af5ecbe711a53b41

jimmynotjim commented:
1. Images - I'm actually uncertain how that works. @kave or @kurtw can you chime in here?
2. `global_dict` - Yeah, I was never really in love with that. I think we're better off with macros when passing around params rather than dirtying thing with global constants.
3. I'm not sure why the homepage would be the only page, especially because we're not using anything dynamic there, it's all static. Maybe it's something in the nav or footer, I'll pull down your fork and take a look a bit later.

sebworks commented:
@jimmynotjim, it was in the `social-media` macro. I pushed a fix for that to this branch.

kave commented:
@richaagarwal So for the hero image i think the simplest solution for now is to include a conditional for image urls passed in by the macro.

```
'macro_image': {
     'url': '/static/img/applying-video-screenshot.png'
}
```
```
{% if macro_image %}
            <div class="hero_image"
                 style="background-image: url( {{ macro_image.url }} );">
        {% endif %}
```

because the `image()` function being currently used is wagtail specific and looks for uploaded images by the user. Since this is not the case a conditional would be more appropriate until we finally transition these pages to wagtail.

richaagarwal commented:
@sebworks thank you!!

richaagarwal commented:
@kave cool, I removed the line with the `image()` function, but left the naming the same.

richaagarwal commented:
I took care of @anselmbradford 's comments. Other than this https://github.com/cfpb/cfgov-refresh/pull/1393#issuecomment-182231900 (@kave, @kurtw ), this is ready for review again!

kave commented on cfgov/jinja2/v1/_includes/organisms/main-contact-info.html:
```diff
@@ -39,52 +39,38 @@
-    {% if value.contact %}
-        <div class="o-main-contact-info">
-            <h2>{{ value.contact.heading }}</h2>
 
```
wrong section was removed. we want to get rid of the macro and leave the `value` related code intact. if the macro is being used else where then leave template as is and make a TODO note.

kave commented on cfgov/jinja2/v1/_includes/molecules/expandable.html:
```diff
@@ -77,41 +74,5 @@
-{% else %}
-    {{ expandable(value) }}
-{% endif %}
+
```
need to add an if conditional 
```
{% if value %}
    {{ expandable(value) }}
{% endif %}
```
or it breaks pages that aren't currently using expandables due to imports e.g. Browse basic template: `{% import 'molecules/expandable.html' as expandable %}`

kave commented on cfgov/jinja2/v1/_includes/organisms/expandable-group.html:
```diff
@@ -23,33 +23,6 @@
-{% endif %}
-{% set value = value or global_dict.expandable_group %}
-
 <div class="o-expandable-group
```
same as my above expandables comment we need a conditional here for value or it breaks when being imported on the page template level. you can test this scenario by creating a `Browse Page` and trying to view the draft.
```
{% if value %}
```

kave commented on cfgov/jinja2/v1/sublanding-page/index.html:
```diff
@@ -1,27 +1,5 @@
-{% endif %}
-
 {% import 'molecules/related-posts.html' as related_posts with context %}
 {% import 'organisms/post-preview.html' as post_preview %}
```
since prototype data has been removed these imports
```
{% import 'organisms/post-preview.html' as post_preview %}
{% import 'organisms/sidebar-contact-info.html' as sidebar_contact_info %}
{% import 'organisms/main-contact-info.html' as main_contact_info %}
```
are unncessary and causing page errors

kave commented on cfgov/jinja2/v1/_includes/templates/content-sidebar.html:
```diff
@@ -0,0 +1,19 @@
+{% for block in page.sidefoot %}
```
since we've moved this code to its own template.

We'll need to include any imports it depends on like. for now it only looks like:
`{% import 'molecules/related-posts.html' as related_posts with context %}` is necessary

kave replied:
@richaagarwal

richaagarwal replied:
@kave sorry - meant to respond to this earlier. I can add it back in, but I thought the context is included?  If not, shouldn't something be breaking?

kave commented on cfgov/cfgov/urls.py:
```diff
@@ -232,9 +232,6 @@
 
-# TODO: Remove prototype landing page routes when all organisms and molecules have been implemented elsewhere.
-if settings.DEBUG :
-    urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
```
those urls have been removed but we're still going to need this particular pattern for images.
```
if settings.DEBUG :		
    urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)		
``

richaagarwal replied:
:+1:

kave commented on cfgov/jinja2/v1/_layouts/base.html:
```diff
@@ -216,8 +216,6 @@
         {% else %}
             {% include '/js/routes/' + request.path[1:] + 'index.js' ignore missing %}
         {% endif %}
-    {% else %}
```
we're still going to need this else conditional to handle legacy pages.. unless there was a reason you removed it?

kave commented:
:+1: Do believe we're seeing the light at the end of the tunnel lol. ran locally all tests pass. beside 2 final comments we can merge this after @sebworks & @KimberlyMunoz PR's go through (acceptance test fixes & form removals respectively)
